src/libraries/System.Linq/src/System/Linq/Enumerable.SizeOpt.cs

```diff
@@ -8,5 +8,10 @@ namespace System.Linq
     public static partial class Enumerable
     {
         public static IEnumerable<TResult> Empty<TResult>() => Array.Empty<TResult>();
+
+        private static IEnumerable<TResult>? GetEmptyIfEmpty<TSource, TResult>(IEnumerable<TSource> source) =>
+            ReferenceEquals(source, Array.Empty<TSource>()) ?
+                Array.Empty<TResult>() :
+                null;
```

Comment:
Why not something like?
```suggestion
            source is TResult[] { Length: 0 } ? source  : null;
```

Reply:
Simply to make the check as cheap as possible while still catching the majority case.  I can change it if you'd prefer.  Note that `source is TSource[]` can be more expensive than it looks, especially when variance is involved (e.g. `source is int[]` is true if source is a `uint[]`, so it's not always just a single cmp).

Reply:
Just curious about the choice. Perhaps add a comment explaining the rationale?

Reply:
I'll change it just to keep it simple.  This is in the size-optimized build, anyway.